code/mission/missionparse.cpp

```diff
@@ -8131,31 +8126,17 @@ void mission_bring_in_support_ship( object *requester_objp )
 
 	vm_set_identity( &(pobj->orient) );
 
-	// *sigh*.  Gotta get the ship class.  For now, this will amount to finding a ship in the ship_info
-	// array with the same team as the requester of type SIF_SUPPORT.  Might need to be changed, but who knows
-
-	// Goober5000 - who knew of the SCP release? ;) only determine ship class if not set by SEXP
+	// If the support ship class was set via SEXP, use it.
+	// Otherwise, look it up by the requester's species.
 	pobj->ship_class = The_mission.support_ships.ship_class;
-	if (pobj->ship_class < 0)
-	{
-		requester_species = Ship_info[requester_shipp->ship_info_index].species;
-
-		// 5/6/98 -- MWA  Don't need to do anything for multiplayer.  I think that we always want to use
-		// the species of the caller ship.
-
-		i = -1;
-		// get index of correct species support ship
-		for (auto it = Ship_info.cbegin(); it != Ship_info.cend(); ++it) {
-            if ((it->species == requester_species) && (it->flags[Ship::Info_Flags::Support])) {
-				i = (int)std::distance(Ship_info.cbegin(), it);
-				break;
-			}
-		}
-
-		if ( i != -1 )
-			pobj->ship_class = i;
-		else
+	if (pobj->ship_class < 0) {
+		int requester_species = Ship_info[requester_shipp->ship_info_index].species;
+		int ship_class = Species_info[requester_species].support_ship_index;
+		if ( ship_class >= 0 ) {
+			pobj->ship_class = ship_class;
+		} else {
 			Int3();				// BOGUS!!!!  gotta figure something out here
```

Comment:
While you're here, I suggest replacing this `Int3()` with a `Warning()`.

Reply:
Done.

Reply:
Ah, that's even better. 👍